### DISCUSSION TITLE
Can't change access role of collaborator

### DISCUSSION THREAD

**Discussion by craigoscore** [community/community#77312] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I would like to change the role of a collaborator for one of my repos... Normally I see a Role dropdown, but for whatever reason it's not there. See attached screenshot.

I have the ability to add new collaborators, but still not able to set their roles.

I am using GitHub Free, this is a private repo. 

![image](https://github.com/community/community/assets/81355412/2b368453-29c5-4bac-911a-413aa14059b5)

**Comment by CheetahCodes21** [comment, marked as answer]
The ability to change collaborator roles is currently not available for private repositories on GitHub Free. This feature is only available for private repositories on GitHub Team and Enterprise Cloud plans.

 If you need to change a collaborator's role on a private repository, you will need to upgrade your GitHub plan to Team or Enterprise Cloud.

If you are unable to upgrade your GitHub plan at this time, you can work around the issue by temporarily making the repository public. Once the repository is public, you will be able to change the collaborator's role. Once you have changed the role, you can make the repository private again.

 Please note that making a private repository public will make it accessible to anyone on the internet. This means that anyone will be able to view the repository's code and make changes to it. If you are concerned about the security of your code, you should not make the repository public.

 I hope this helps!

**Reply by BogdanRemusPintilie** [reply]
Despite making my repository public, I still can not find any dropdown for each collaborator to grant them admin rights.

If what you say is true, anyone could switch between public and private and doge the necessity for an upgraded plan.

Please correct me if I am wrong.

**Reply by CheetahCodes21** [reply]
You are correct. Granting individual admin rights to collaborators in a public GitHub repository is not possible. Public repositories come with specific access levels, and the highest level for collaborators is "Write," which allows them to push code, create branches, and open pull requests. However, they cannot manage access itself, which includes adding or removing other collaborators or changing access levels.

Therefore, switching a repository from public to private to bypass the necessity for an upgraded plan wouldn't work. While the access control becomes more granular in a private repository, allowing you to grant individual admin rights, the upgrade to a paid plan is still required for functionalities like private collaborators, unlimited private repositories, and other features unavailable in the free tier.

Here's a recap of the key points:

* **Public repositories:** Limited access levels, collaborators can have "Write" access but not admin rights.
* **Private repositories:** Granular access control, admins can grant individual admin rights to collaborators.
* **Free vs. paid plans:** Free plan limits some features like private repositories and collaborator types. Paid plans offer these features.

As for the dropdown menu you mentioned, there isn't one specifically for assigning admin rights to collaborators in any type of repository. Access control is managed through the "Manage access" settings in your repository, where you can grant different roles like "Write" or "Admin" to individual collaborators.

I hope this clarifies the situation. If you have any further questions or would like to discuss it in more detail, feel free to ask!

**Reply by DragonGong** [reply]
My repository was private before, but I changed it to public now , I still don't see the ability to change the roles of collaborator, why is this?
is this  a bug?

**Reply by firmai** [reply]
Same here, very annoying

**Reply by johannes-do** [reply]
I dont get why public repos wouldnt have that feature. I want to grant some collaborators admin rights so they can add others, create releases, merge branches without code review etc.

**Reply by FilterUnfiltered** [reply]
I have the same issue, except mine has always been public? My public repo has no option to change permissions. This is the only case I can find where it seems to almost look the same, but according to these replies I should only have this issue if the repo is private, which it isn't.

**Reply by mazenemam19** [reply]
I'm still facing this issue :(

**Reply by psh9508** [reply]
I think there is something wrong with this function. It is still here in 2025.
It's a public repository, and I'm the one who created it. Did I miss anything?
<img width="395" height="112" alt="image" src="https://github.com/user-attachments/assets/d0951478-f28a-4095-b37b-5183151f4314" />

**Reply by datlissschen** [reply]
Hey, I still have the same problem although my repo is actually public.

**Reply by edutechab** [reply]
It seems the ability to change role for collaborators has been removed for both public and private users on FREE PLAN
